specs/Menu.md

```diff
@@ -1,731 +1,672 @@
+<div role="menu">
+  <div role="menuitem" tabindex="0">Option 1</div>
+  <div role="separator"></div>
+  <div role="menuitem" tabindex="-1">Section Option 1</div>
+  <div role="menuitem" tabindex="-1">Section Option 2</div>
+  <div role="menuitem" tabindex="-1">Section Option 3</div>
+  <div role="separator"></div>
+  <!-- TODO submenu -->
+</div>
+```
 
-By default Focus should go to the tab stops and it's up to each menuItem to decide if it has tabindex=0. An example of a menuItem that shouldn't be focusable would be a Divider.
+### Standlone
 
-However for the menu we should provide a focus wrapper that is able to track focus using role="menuitem" to determine what item should get focused.
+```typescript
+const [open] = React.useState(false);
 
-The menu does not need to provide a native way to be controlled without getting focus directly. Like above, there should be a simple utility wrapper that could be written to implement this.
+const menu = (
+  <CustomSurface open={open}>
+    <MenuList>
+      <MenuItem>Option 1</MenuItem>
+      <MenuItem>Option 2</MenuItem>
+      <MenuItem>Option 3</MenuItem>
+    <MenuList>
+  <CustomSurface>
+)
+```
 
-##### On Menu Click
+```html
+<!-- expected DOM output  -->
+<div role="menu">
+  <div role="menuitem" tabindex="0">Option 1</div>
+  <div role="menuitem" tabindex="-1">Option 2</div>
+  <div role="menuitem" tabindex="-1">Option 3</div>
+</div>
+```
 
-Some implementations of menu have a general `onClick` function that is applied to all items as a way of getting more information.
+### Selection
+
+```typescript
+const trigger = <button> Open menu </button>
+const [selectedItems, setSelectedItems] = React.useState([]);
+
+// basic checkbox example
+const menuCheckbox = (
+  <Menu
+    selectedItems={selectedItems}
+    onSelectionChange={setSeelctedItems}
+    trigger={trigger}
+  >
+    <MenuItem index={1}>Option 1</MenuItem>
+    <MenuItem index={2}>Option 2</MenuItem>
+    <MenuItem index={3}>Option 3</MenuItem>
+  <Menu>
+)
+
+// leverage MenuSection for different selection groups
+const menuSelectableSections = (
+  <Menu
+    selectedItems={selectedItems}
+    onSelectionChange={setSeelctedItems}
+    trigger={trigger}
+  >
+    <MenuSection kind="checkbox">
+      <MenuItem index={1}>Option 1</MenuItem>
+      <MenuItem index={2}>Option 2</MenuItem>
+      <MenuItem index={3}>Option 3</MenuItem>
+    </MenuSection>
+    <MenuSection kind="radio">
+      <MenuItem index={4}>Option 1</MenuItem>
+      <MenuItem index={5}>Option 2</MenuItem>
+      <MenuItem index={6}>Option 3</MenuItem>
+    </MenuSection>
+  <Menu>
+)
+```
 
-Recommendation: There is not a need to have an `onClick` that is called whenever an item is clicked. Instead the root `menu` should take an `onClick` that will fire if a child's on click does not prevent default.
+```html
+<!-- expected DOM output for basic checkbox  -->
+<div role="menu">
+  <div role="menuitemcheckbox" tabindex="0" aria-checked="true">Option 1</div>
```

Comment:
When are the `role` and `aria-checked` added?

Reply:
This should be handled by each `MenuItem` from proposed context that would be used in the `Menu`

I definitely do need to add more detail about what context should do that's a good point